src/Sarif/Writers/SarifLogger.cs

```diff
@@ -278,9 +287,80 @@ public void Log(IRule rule, Result result)
                 Rules[result.RuleKey ?? result.RuleId] = rule;
             }
 
+            CaptureFilesInResult(result);
```

Comment:
This is really nice. Is this the only way to populate `run.files`? Is it possible that the SDK client could populate `Run.Files` explicitly, and then have it overwritten (or added on to) by this code? Does there need to be a way to turn this off?